packages/block-editor/src/components/link-control/README.md

```diff
@@ -71,3 +71,60 @@ Whether to present initial suggestions immediately.
 - Required: No
 
 If passed as either `true` or `false`, controls the internal editing state of the component to respective show or not show the URL input field.
+
+
+### createSuggestion
+
+- Type: `function`
+- Required: No
+
+Used to handle the dynamic creation of new suggestions within the Link UI. When
+the prop is provided, an option is added to the end of all search
+results requests which when clicked will call `createSuggestion` callback
+(passing the current value of the search `<input>`) in
+order to afford the parent component the opportunity to dynamically create a new
+link `value` (see above).
+
+This is often used to allow on-the-fly creation of new entities (eg: `Posts`,
+`Pages`) based on the text the user has entered into the link search UI. For
+example, the Navigation Block uses this to create Pages on demand.
+
+When called, `createSuggestion` may return either a new link `value` or a `Promise` which resolves to a
+new link `value` (see above) with the shape:
```

Comment:
A suggestion is technically not the same as a value, as noted in previous reviews and discussion.

Reply:
Let me review the types again and make sure I fully understand the definitions.

Reply:
Updated to clarify that it is a `suggestion` and not a `value`.